yt/yt/core/misc/async_slru_cache-inl.h

```diff
@@ -1252,11 +1358,22 @@ void TAsyncSlruCacheBase<TKey, TValue, THash>::TShard::OnOlderUpdated(i64 deltaC
     Parent->OlderWeightCounter_ += deltaWeight;
 }
 
+template <class TKey, class TValue, class THash>
+void TAsyncSlruCacheBase<TKey, TValue, THash>::TShard::OnCookieUpdated(i64 deltaCount, i64 deltaWeight)
+{
+    Parent->CookieSizeCounter_ += deltaCount;
+    Parent->CookieWeightCounter_ += deltaWeight;
+}
+
 template <class TKey, class TValue, class THash>
 void TAsyncSlruCacheBase<TKey, TValue, THash>::NotifyOnTrim(
     const std::vector<TValuePtr>& evictedValues,
-    const TValuePtr& insertedValue)
+    const TValuePtr& insertedValue,
+    i64 weightDelta)
 {
+    if (weightDelta) {
```

Comment:
``> 0`` here as well

Reply:
!= 0 then - it could be called with negative delta from EndInsert